gstreamer-sdp/src/s_d_p_message.rs

```diff
@@ -1,11 +1,440 @@
+// Copyright (C) 2018 Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use ffi;
+use glib;
+use glib_ffi;
+
+use auto::SDPResult;
+use auto::MikeyMessage;
+use s_d_p_attribute::SDPAttribute;
+use s_d_p_bandwidth::SDPBandwidth;
+use s_d_p_connection::SDPConnection;
+use s_d_p_key::SDPKey;
+use s_d_p_origin::SDPOrigin;
+use s_d_p_time::SDPTime;
+use s_d_p_zone::SDPZone;
+
+glib_wrapper! {
+    pub struct SDPMessage(Boxed<ffi::GstSDPMessage>);
+
+    match fn {
+        copy => |ptr| gobject_ffi::g_boxed_copy(ffi::gst_sdp_message_get_type(), ptr as *mut _) as *mut ffi::GstSDPMessage,
+        free => |ptr| gobject_ffi::g_boxed_free(ffi::gst_sdp_message_get_type(), ptr as *mut _),
+        get_type => || ffi::gst_sdp_message_get_type(),
+    }
+}
 
 impl SDPMessage {
     pub fn new() -> SDPMessage {
         assert_initialized_main_thread!();
         unsafe {
             let msg = glib_ffi::g_malloc(mem::size_of::<ffi::GstSDPMessage>());
-            let _ = ffi::gst_sdp_message_new(msg);
+            ffi::gst_sdp_message_new(msg);
+            from_glib_full(msg)
+        }
+    }
+
+	pub fn init() -> SDPMessage {
```

Comment:
So after I changed this to a constructor, I noticed that this is doing essentially the same thing as `new()`.

Reply:
Yeah, init() is unneeded in this case